python-package/xgboost/spark/core.py

```diff
@@ -755,7 +757,11 @@ def _fit(self, dataset):
             k: v for k, v in train_call_kwargs_params.items() if v is not None
         }
         dmatrix_kwargs = {k: v for k, v in dmatrix_kwargs.items() if v is not None}
-        use_qdm = booster_params.get("tree_method", None) in ("hist", "gpu_hist")
+
+        # If cuDF is not installed, then using DMatrix instead of QDM,
+        # because without cuDF, DMatrix performs better than QDM.
+        use_qdm = CUDF_INSTALLED and \
```

Comment:
I'm still confused why do we still need to check the cudf on the driver side? If we have packed all the xgboost dependencies including CUDF into a zip and specify it by `--py-files` when spark-submitting. Even if the driver doesn't have cudf installed, the training task can still use qdm.

Reply:
OK I can move the check to worker side. :)

Reply:
@wbo4958 @trivialfis Updated!